Python/ceval.c

```diff
@@ -4719,19 +4719,7 @@ _PyEval_EvalFrameDefault(PyThreadState *tstate, _PyInterpreterFrame *frame, int
             PyObject *func, *callargs, *kwargs = NULL, *result;
             if (oparg & 0x01) {
                 kwargs = POP();
-                if (!PyDict_CheckExact(kwargs)) {
```

Comment:
Could you add an assert that `kwargs` is a dict and a short comment explaining why it must be a dict.

Reply:
Assert is already there https://github.com/python/cpython/pull/96752/files#diff-c22186367cbe20233e843261998dc027ae5f1f8c0d2e778abfa454ae74cc59deR4723

Right now comment says `// DICT_MERGE is called before this opcode.`
https://github.com/python/cpython/pull/96752/files#diff-c22186367cbe20233e843261998dc027ae5f1f8c0d2e778abfa454ae74cc59deR4722

I think I can improve it!